From 055862af4062e2feecd445fddfa87714c53edb50 Mon Sep 17 00:00:00 2001
From: Peter Jonas <shoogle@users.noreply.github.com>
Date: Wed, 15 Sep 2021 20:32:54 +0100
Subject: [PATCH] Restore missing localizations on Linux

The INSTALL_PREFIX variable in src/audacity_config.h.in was being
passed into the program as an empty string. The file is configured
in the top-level CMakeLists.txt but INSTALL_PREFIX is defined in
src/CMakeLists.txt. CMake variables are not passed to the parent scope
unless explicitly requested.

Partially fix #1382. Allows Audacity to find the translations when they
are at the default location specified by INSTALL_PREFIX, but not if
they have been moved elsewhere. This works for distro builds but it's
not enough to get the AppImage working again.
---
 CMakeLists.txt     | 2 +-
 src/CMakeLists.txt | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e56220fa73..49303b58e00 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -316,7 +316,7@@ else()
 endif()
 
 set( _DEST "${_DESTDIR}" )
-set( _PREFIX "${CMAKE_INSTALL_PREFIX}" )
+set( INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" )
 set( _LIBDIR "${CMAKE_INSTALL_LIBDIR}" )
 set( _DATADIR "${CMAKE_INSTALL_DATADIR}" )
 set( _PKGLIB "${_LIBDIR}/audacity" )
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e61eba8dcbb..7938f9f42e9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1122,7 +1122,6 @@ list( APPEND LIBRARIES
 )
 
 set( BUILDING_AUDACITY YES )
-set( INSTALL_PREFIX "${_PREFIX}" )
 set( PKGLIBDIR "${_PKGLIBDIR}" )
 set( LIBDIR "${CMAKE_INSTALL_FULL_LIBDIR}" )
 set( HAVE_GTK ${GTK_FOUND} )
